### PR TITLE
Mac: Enable signing with self signed cert

### PR DESCRIPTION
**Pull request by ann0see**
**Short description of changes**
Enables the CI to use a code signing certificate signed by a non Apple CA to sign development builds. 

grafik
grafik

CHANGELOG: 

**Context: Fixes an issue?**

Fixes: https://github.com/jamulussoftware/jamulus/issues/2924

**Does this change need documentation? What needs to be documented and how?**

Probably not. 

**Status of this Pull Request**

Ready for review (and test on Apple Silicon, a repo without the respective secrets set and a repo with the real apple secrets set (@emlynmac 's repo))

**What is missing until this pull request can be merged?**

Still needs some (external testing). Artifacts are building on my repo: https://github.com/ann0see/jamulus/actions/runs/3352547263/jobs/5554760442
## Checklist

-  [x] I've verified that this Pull Request follows the [general code principles](https://github.com/jamulussoftware/jamulus/blob/master/CONTRIBUTING.md#jamulus-projectsource-code-general-principles)
-  [x] I tested my code and it does what I want (on my repo)
-  [x] My code follows the [style guide](https://github.com/jamulussoftware/jamulus/blob/master/CONTRIBUTING.md#source-code-consistency) 
-  [x] I waited some time after this Pull Request was opened and all GitHub checks completed without errors. 
-  [x] I've filled all the content above

AUTOBUILD: Please build all targets

### CODE REVIEW

File: .github/autobuild/mac.sh [hunk 1]
```diff
@@ -41,14 +41,27 @@ prepare_signing() {
     [[ -n "${MACOS_CERTIFICATE:-}" ]] || return 1
     [[ -n "${MACOS_CERTIFICATE_ID:-}" ]] || return 1
     [[ -n "${MACOS_CERTIFICATE_PWD:-}" ]] || return 1
-    [[ -n "${NOTARIZATION_PASSWORD:-}" ]] || return 1
     [[ -n "${KEYCHAIN_PASSWORD:-}" ]] || return 1
 
+    # Check for notarization (not wanted on self signed build)
+    if [ -z "${NOTARIZATION_PASSWORD}" ]; then
```
Comment (hoffie) [inline .github/autobuild/mac.sh:47]:
```suggestion
    if [[ -z "${NOTARIZATION_PASSWORD}" ]]; then
```

File: .github/autobuild/mac.sh [hunk 2]
```diff
@@ -41,14 +41,27 @@ prepare_signing() {
...
+        echo "Notarization password not found or empty. This suggests we might run a self signed build."
+        if [ -z "${MACOS_CA_PUBLICKEY}" ]; then
```
Comment (hoffie) [inline .github/autobuild/mac.sh:49]:
```suggestion
        if [[ -z "${MACOS_CA_PUBLICKEY}" ]]; then
```

**Review by hoffie — APPROVED** [review]
Minor nit wrt `[`usage. Other than that, looks good. Thanks!

As you say, it should receive proper testing, especially of release builds. I think you still have access to @emlynmac's repo as well and could build a fake 0.something version for testing?

**Comment by ann0see** [comment]
I've pushed to emlyns repo, but my macOS VM is currently unavailable, so I can't test.

**Comment by ann0see** [comment]
Ok. Tested yesterday, and it seems to be signed.

**Comment by ann0see** [comment]
@hoffie I probably did some mistake during the rebase (again...). However, I think it's ok now. Please review again, nevertheless.

**Comment by ann0see** [comment]
@pljones @hoffie could you please review this?

**Comment by pljones** [comment]
I've no idea what any of the changes mean.  It's not affecting anything other than MacOS and none of the builds broke, so I'm happy with it.

**Comment by emlynmac** [comment]
Can we make sure this actually does what is expected?
All of the artifacts have expired, so not possible to test.
I'd be surprised if a self signed binary works on a macOS platform TBH.

**Comment by ann0see** [comment]
> I'd be surprised if a self signed binary works on a macOS platform TBH.

They do still show a warning but the point is that cert signing problems would show up in production already.

**Comment by emlynmac** [comment]
> > I'd be surprised if a self signed binary works on a macOS platform TBH.
> 
> They do still show a warning but the point is that cert signing problems would show up in production already.

If this is purely to validate the process, then the signed artifact should not be published.

**Comment by ann0see** [comment]
I’d there any downside of a self signed test version?

**Comment by emlynmac** [comment]
> I’d there any downside of a self signed test version?

If it's published and people try to use it, then issues might be raised in confusion.

**Comment by ann0see** [comment]
Yes, but the same would be true for unsigned builds. This would just affect PR builds and maybe the legacy build. Release builds would be built on your repo.

**Comment by pljones** [comment]
Hi @ann0see, is there any progress here?

**Comment by ann0see** [comment]
There's still a Test and OK by Emlyn outstanding.

**Review by pljones — APPROVED** [review]
Looks syntactically correct.

**Comment by ann0see** [comment]
Ok. Now we need to add a certificate.

**Pull request merged**